rhino/src/main/java/org/mozilla/javascript/regexp/NativeRegExp.java

```diff
@@ -1374,6 +1687,24 @@ private static boolean flatNIMatcher(
         return true;
     }
 
+    private static boolean flatNIMatcherBackward(
```

Comment:
It doesn't look like this function gets called in tests

Reply:
Ah, these are missing tests indeed. Added them.